pkg/knuu/knuu.go

```diff
@@ -42,9 +42,9 @@ type Options struct {
 	K8s          k8s.KubeManager
 	TestScope    string
 	ImageBuilder builder.Builder
-	Minio        *minio.Minio
-	Timeout      time.Duration
+	MinioEnabled bool
```

Comment:
I don't think we need the boolean. We can just have the `Minio` instance. 
Since it isn't a required service, developers can pass in `nil`. 
If they plan to use it, they just initialize the minio client and pass that in with the options.

Reply:
Ideally we could do it, then we need to also ask user to pass k8sClient as well since we need it for deploying minio.
Also we might also question the enabling proxy which is also a boolean.
So wdyt?

Reply:
I think also receiving k8s client from user makes more sense

Reply:
The k8 client is already passed in by the user, so the user has that to initialize the minio client. 

It also causes less of a diff, since there isn't a new option. It is just the change in default behavior that is impacted. 

For how the `New` method is currently structured, I'd recommend accepting the actual clients vs flags. I wouldn't change the `ProxyEnabled` flag just yet though. We should review the New method again after this PR is merged and see if there are any other updates we want to make as part of the breaking release.

Reply:
Ok pushed some changes. k8sClient was accepted as an option and if it was not passed, a default client would have been initiated. Now, user has to pass it as well. I think now it is in a pretty good shape. thanks for pointing it out.
Agreed that the proxy is something more embedded in knuu and refactoring its option requires more thought.